gprofiler/profilers/java.py

```diff
@@ -124,6 +126,53 @@ def get_ap_log(self) -> str:
         return self._ap_log
 
 
+_JAVA_VERSION_TIMEOUT = 5
+
+
+def get_java_version(process: Process, stop_event: Event) -> str:
+    nspid = get_process_nspid(process.pid)
+
+    # this has the benefit of working even if the Java binary was replaced, e.g due to an upgrade.
+    # in that case, the libraries would have been replaced as well, and therefore we're actually checking
+    # the version of the now installed Java, and not the running one.
+    # but since this is used for the "JDK type" check, it's good enough - we don't expect that to change.
+    # this whole check, however, is growing to be too complex, and we should consider other approaches
+    # for it:
+    # 1. purely in async-profiler - before calling any APIs that might harm blacklisted JDKs, we can
+    #    check the JDK type in async-profiler itself.
+    # 2. assume JDK type by the path, e.g the "java" Docker image has
+    #    "/usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java" which means "OpenJDK". needs to be checked for
+    #    other JDK types.
+    java_path = f"/proc/{nspid}/exe"
+
+    def _run_java_version() -> "CompletedProcess[bytes]":
+        return run_process(
+            [
+                java_path,
+                "-version",
+            ],
+            stop_event=stop_event,
+            timeout=_JAVA_VERSION_TIMEOUT,
+        )
+
+    # doesn't work without changing PID NS as well (I'm getting ENOENT for libjli.so)
+    # Version is printed to stderr
+    return run_in_ns(["pid", "mnt"], _run_java_version, process.pid).stderr.decode().strip()
+
+
+class JavaMetadta(ApplicationMetadata):
+    @classmethod
+    def make_application_metadata(cls, process: Process, stop_event: Event) -> Dict[str, Any]:
+        version = get_java_version(process, stop_event)
+        # libjvm elfid
+        libjvm_elfid = get_mapped_dso_elf_id(process, "/libjvm")
+
+        metadata = {"java_version": version, "libjvm_elfid": libjvm_elfid}
```

Comment:
You mean `java_elfid`? Java is always a small binary that loads `libjvm.so`, we've never cared about it on its own.
In Python, sometimes there's no `libpython` so Python is interesting. I decided to retrieve both libpython and python in all cases, to avoid any chance for confusion (passing None if not found).

I'll document to clarify.